src/sysroot.rs

```diff
@@ -303,6 +301,26 @@ pub struct Blueprint {
     stages: BTreeMap<i64, Stage>,
 }
 
+trait AsTableMut {
+    fn as_table_mut<F, R>(&mut self, on_error_path: F) -> Result<&mut Table>
+    where
+        F: FnOnce() -> R,
+        R: ::std::fmt::Display;
+}
+
+impl AsTableMut for Value {
+    fn as_table_mut<F, R>(&mut self, on_error_path: F) -> Result<&mut Table>
+    where
+        F: FnOnce() -> R,
+        R: ::std::fmt::Display,
+    {
+        match self {
+            Value::Table(table) => Ok(table),
+            _ => Err(format!("Xargo.toml: `{}` must be a table", on_error_path()).into()),
+        }
+    }
+}
```

Comment:
> This is new, isn't it?

Yes

> It doesn't seem to simplify things overall

What do you mean? It's used 3 times. Do you think it'd be "simpler" if I just take a `String` with the TOML path instead of a closure?

Reply:
I mean compared to the prior version that just had a local helper method, not a new trait.

But I guess I don't care strongly either way. Please add a doc comment for the method though.

Reply:
By local helper method, I think you're referring to `table_entry`? That does something else (and is still there).

Reply:
Yeah, `table_entry` used to have this embedded.